src/plugins/intel_gpu/src/graph/impls/ocl/scaled_dot_product_attention.cpp

```diff
@@ -335,12 +338,34 @@ struct scaled_dot_product_attention_impl : multi_stage_primitive<scaled_dot_prod
             data_inputs_num++;
         }
 
-        params.conf = get_sdpa_configuration(impl_param);
+        auto extend_order_in_num_heads_dim = [](const std::vector<int64_t>& order, size_t rank = 4) {
+            if (order.size() == rank) {
+                return order;
+            }
+
+            std::vector<int64_t> extended_order(rank, 0);
+            extended_order[1] = 1;
```

Comment:
```suggestion
            const size_t head_num_dim = 1;
            extended_order[head_num_dim] = 1;
            // For 3D dimension, extend it to 4D with adding a 1 for head_num_dim
            for (size_t i = 0; i < rank; ++i) {
                if (i < head_num_dim)
                    extended_order[i] = order[i];
                else if (i == head_num_dim)
                    continue;
                // i > head_num_Dim    
                   extended_order[i] = order[ i - 1] + 1;
           }
            
```
nit but I'd like to suggest a little bit more descriptive way

Reply:
updated as suggested. however `j` increment needs to be placed inside `else` for it to work properly